kb/2024-07-22/harvester_cluster_shutdown_and_restart.md

```diff
@@ -0,0 +1,576 @@
+| https://192.168.122.141:2379 | c70780b7862269c9 |   3.5.9 |   34 MB |      true |      false |        47 |    6538833 |            6538833 |        |
+| https://192.168.122.142:2379 | db04095b49eb5352 |   3.5.9 |   34 MB |     false |       true |        47 |    6538833 |            6538833 |        |
+| https://192.168.122.143:2379 | a21534d02463b347 |   3.5.9 |   34 MB |     false |       true |        47 |    6538833 |            6538833 |        |
++------------------------------+------------------+---------+---------+-----------+------------+-----------+------------+--------------------+--------+
+```
+
+To now, the `etcd` has three running instances and the leader is located on the `control-plane` node.
+
+:::info important
+
+Write down the information of those nodes like name, ip, and the leader. Ideally give them a sequence like 1, 2, 3.
+
+:::
+
+Shutdown the two `IS LEADER == false` nodes one by one.
+
+```
+harvester-node-shell$ sudo -i shutdown
+```
+
+### 4.3 Shutdown the Last Control-plane Node
+
+Shutdown the last `IS LEADER == true` node. Remember its physical information for restarting it first in the steps below.
+
+```
+harvester-last-node-shell$ sudo -i shutdown
+```
+
+## 5. Restart
+
+If the Harvester cluster has been moved to a new location, or has been off for days, or your infrastructure has changes, check and test the [network stability](#12-keep-network-stability).
+
+### 5.1 Restart the Control-plane Nodes and the Witness Node
+
+The first step is to start those `etcd` located nodes one after another.
+
+#### Restart the Leader Control-plane Node
+
+Power on the [last shutdown node](#43-shutdown-the-last-control-plane-node) first. After about three minutes, continue the next step.
+
+When you check the `etcd` pod log on this node, the following message may be observed.
+
+```
+sent MsgPreVote request to db04095b49eb5352 at term 5
+
+"msg":"prober detected unhealthy status","round-tripper-name":"ROUND_TRIPPER_RAFT_MESSAGE","remote-peer-id":"db04095b49eb5352","rtt":"0s","error":"dial tcp 192.168.122.142:2380: connect: no route to host"
+
+```
+
+The `etcd` is wating for the other two members to be online and then vote a leader.
+
+#### Restart the Rest of Control-plane Nodes and the Witness Node
+
+Power on the rest nodes which also hosted the `etcd` pod before.
+
+Wait until all the three control-plane nodes or possibly two control-plane and one witness nodes are `Ready`.
+
+From CLI:
+
+```
+harvester$ kubectl get nodes -A
+NAME     STATUS   ROLES                       AGE   VERSION
+harv41   Ready    control-plane,etcd,master   54d   v1.27.10+rke2r1
+harv42   Ready    control-plane,etcd,master   54d   v1.27.10+rke2r1
+harv43   Ready    control-plane,etcd,master   54d   v1.27.10+rke2r1
+
+```
+
+The `etcd` forms a quorum and can tolerant the failure of one node.
+
+:::note
+
+If the `embedded rancher` was not scaled down before, this step can also be:
```

Comment:
Based on the previous decision (see above), this should also be adjusted here.
E.g. embedded `Rancher` or embedded `rancher`, ...